src/winrtble/peripheral.rs

```diff
@@ -81,13 +86,66 @@ impl Peripheral {
             properties.manufacturer_data = manufacturer_data
                 .into_iter()
                 .map(|d| {
-                    let company_id = d.company_id().unwrap();
+                    let manufacturer_id = d.company_id().unwrap();
                     let buffer = d.data().unwrap();
                     let reader = DataReader::from_buffer(&buffer).unwrap();
                     let len = reader.unconsumed_buffer_length().unwrap() as usize;
                     let mut data = vec![0u8; len];
                     reader.read_bytes(&mut data).unwrap();
-                    (company_id, data)
+
+                    // Emit event of newly received advertisement
+                    self.adapter.emit(CentralEvent::ManufacturerAdvertisement {
+                        address: self.address,
+                        manufacturer_id,
+                        data: data.clone(),
+                    });
+
+                    (manufacturer_id, data)
+                })
+                .collect();
+        }
+
+        if let Ok(data_sections) = advertisement.data_sections() {
```

Comment:
It's probably worth adding a comment here that we have to do this because the Windows API for some reason doesn't parse service data for us.

Reply:
I've added a comment, I hope it explains why, well enough

Reply:
Looks good to me.